src/corelibs/U2Algorithm/src/util_msa_consensus/MSAConsensusAlgorithmLevitsky.cpp

```diff
@@ -133,28 +133,27 @@ char MSAConsensusAlgorithmLevitsky::getConsensusChar(const MultipleAlignment& ma
 
     // count local freqs first
     QVarLengthArray<int> localFreqs(256);
-    memset(localFreqs.data(), 0, localFreqs.size() * 4);
-
+    for (int i = 0; i < 256; i++) {
+        localFreqs[i] = 0;
+    }
     int* freqsData = localFreqs.data();
     int nSeq = (seqIdx.isEmpty() ? ma->getRowCount() : seqIdx.size());
     for (int seq = 0; seq < nSeq; seq++) {
         char c = ma->charAt(seqIdx.isEmpty() ? seq : seqIdx[seq], column);
         registerHit(freqsData, c);
     }
-
     // find all symbols with freq > threshold, select one with the lowest global freq
     char selectedChar = U2Msa::GAP_CHAR;
-    int selectedGlobalFreq = nSeq * ma->getLength();
+    double selectedGlobalPercentage = 100;
     int thresholdScore = getThreshold();
-    int minFreq = int(float(nSeq) * thresholdScore / 100);
     for (int c = 'A'; c <= 'Y'; c++) {
-        int localFreq = freqsData[uchar(c)];
-        if (localFreq < minFreq) {
+        double localPercentage = ((double)freqsData[uchar(c)] / (double)nSeq) * 100;
```

Comment:
nit:
1. one (double) is enough.
2. when you work with double you can remove *100 completely and work on 0..1 (share) range.